sql/core/src/test/resources/sql-tests/inputs/pgSQL/union.sql

```diff
@@ -0,0 +1,472 @@
+
+(((SELECT q1 FROM int8_tbl INTERSECT SELECT q2 FROM int8_tbl ORDER BY 1))) UNION ALL SELECT q2 FROM int8_tbl;
+
+SELECT q1 FROM int8_tbl UNION ALL SELECT q2 FROM int8_tbl EXCEPT SELECT q1 FROM int8_tbl ORDER BY 1;
+
+SELECT q1 FROM int8_tbl UNION ALL (((SELECT q2 FROM int8_tbl EXCEPT SELECT q1 FROM int8_tbl ORDER BY 1)));
+
+(((SELECT q1 FROM int8_tbl UNION ALL SELECT q2 FROM int8_tbl))) EXCEPT SELECT q1 FROM int8_tbl ORDER BY 1;
+
+--
+-- Subqueries with ORDER BY & LIMIT clauses
+--
+
+-- In this syntax, ORDER BY/LIMIT apply to the result of the EXCEPT
+SELECT q1,q2 FROM int8_tbl EXCEPT SELECT q2,q1 FROM int8_tbl
+ORDER BY q2,q1;
+
+-- This should fail, because q2 isn't a name of an EXCEPT output column
+SELECT q1 FROM int8_tbl EXCEPT SELECT q2 FROM int8_tbl ORDER BY q2 LIMIT 1;
+
+-- But this should work:
+SELECT q1 FROM int8_tbl EXCEPT (((SELECT q2 FROM int8_tbl ORDER BY q2 LIMIT 1))) ORDER BY 1;
+
+--
+-- New syntaxes (7.1) permit new tests
+--
+
+(((((select * from int8_tbl)))));
+
+-- [SPARK-28557] Support empty select list
+--
+-- Check behavior with empty select list (allowed since 9.4)
+--
+
+-- select union select;
+-- select intersect select;
+-- select except select;
+
+-- check hashed implementation
+-- set enable_hashagg = true;
+-- set enable_sort = false;
+
+-- explain (costs off)
+-- select from generate_series(1,5) union select from generate_series(1,3);
+-- explain (costs off)
+-- select from generate_series(1,5) intersect select from generate_series(1,3);
+
+-- [SPARK-28409] SELECT FROM syntax
+-- [SPARK-27767] Built-in function: generate_series
+select * from range(1,5) union select * from range(1,3);
+select * from range(1,6) union all select * from range(1,4);
+select * from range(1,6) intersect select * from range(1,4);
+select * from range(1,6) intersect all select * from range(1,4);
+select * from range(1,6) except select * from range(1,4);
+select * from range(1,6) except all select * from range(1,4);
+
+-- check sorted implementation
+-- set enable_hashagg = false;
+-- set enable_sort = true;
+
+-- explain (costs off)
+-- select from generate_series(1,5) union select from generate_series(1,3);
+-- explain (costs off)
+-- select from generate_series(1,5) intersect select from generate_series(1,3);
+
+select * from range(1,6) union select * from range(1,4);
+select * from range(1,6) union all select * from range(1,4);
+select * from range(1,6) intersect select * from range(1,4);
+select * from range(1,6) intersect all select * from range(1,4);
+select * from range(1,6) except select * from range(1,4);
+select * from range(1,6) except all select * from range(1,4);
+
+-- reset enable_hashagg;
+-- reset enable_sort;
+
+--
+-- Check handling of a case with unknown constants.  We don't guarantee
+-- an undecorated constant will work in all cases, but historically this
+-- usage has worked, so test we don't break it.
+--
+
+-- SELECT a.f1 FROM (SELECT 'test' AS f1 FROM varchar_tbl) a
```

Comment:
Any reason for commenting out this?

Reply:
We can not fully support char and varchar types.

Reply:
Ur, I see.